kratos/includes/parallel_environment.h

```diff
@@ -142,6 +147,9 @@ class KRATOS_API(KRATOS_CORE) ParallelEnvironment
 
     std::unordered_map<std::string, DataCommunicator::UniquePointer>::iterator mDefaultCommunicator;
 
+    static ParallelEnvironment* mpInstance;
```

Comment:
Why you have not used a Meyer singleton here?

Reply:
I use the pointer to guard against a race condition if the first access even happens to be within an openmp parallel region, as well as to keep track of singleton destruction:
https://github.com/KratosMultiphysics/Kratos/blob/abd79e5d7fba2aaf3cff4b19aacac9d7934e5c76/kratos/sources/parallel_environment.cpp#L89-L113

The object itself is still a static (local) variable. This is all from the "recipe" in Modern C++, and it helped me to catch a bug due to interaction between the `ParallelEnvironment` and `MPIEnvironment` object lifetimes.